main/src/library/Fixpoint/Interpreter.flix

```diff
@@ -76,20 +76,20 @@ namespace Fixpoint {
     }
 
     type alias Database[v: Type, r: Region] = MutMap[RamSym[v], MutMap[Tuple[v], v, r], r]
-    type alias SearchEnv[v] = (Array[Tuple[v], Static], Array[v, Static])
+    type alias SearchEnv[v: Type, r: Region] = (Array[Tuple[v], r], Array[v, r])
 
-    def interpret(r: Region[r], stmt: RamStmt[v]): Database[v, r] & Impure with Order[v], ToString[v] =
+    def interpret(r: Region[r], stmt: RamStmt[v]): Database[v, r] & r with Order[v], ToString[v] =
         interpretWithDatabase(new MutMap(r), stmt)
 
-    def interpretWithDatabase(db: Database[v, r], stmt: RamStmt[v]): Database[v, r] & Impure with Order[v], ToString[v] =
-        notifyPreInterpret(stmt);
+    def interpretWithDatabase(db: Database[v, r], stmt: RamStmt[v]): Database[v, r] & r with Order[v], ToString[v] =
+        let _ = notifyPreInterpret(stmt) as & Pure;
```

Comment:
If the cast is not on the call to notifyPre... but on the whole body of the function then it should not be removed, right?

Reply:
But then of course that cast has to be to r.